javascript/src/main/components/TutorAssignment/TutorAssignmentSelect.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import { Cascader } from "antd";
+import { asHumanQuarter } from "main/utils/quarter";
+
+import {
+  quarterProvider as defaultQuarterProvider,
+  courseProvider as defaultCourseProvider,
+  tutorAssignmentProvider as defaultTutorAssignmentProvider
+} from "main/services/selectorSupport.js"
+
+export const selectorIndices = {
+  QUARTER: 0,
+  COURSE: 1,
+  TUTOR_ASSIGNMENT: 2
+};
+
+export default function TutorAssignmentSelect(props) {
+  const {
+    quarterProvider = defaultQuarterProvider,
+    courseProvider = defaultCourseProvider,
+    tutorAssignmentProvider = defaultTutorAssignmentProvider,
+    onChange,
+    ...otherProps
+  } = props;
+  const [options, setOptions] = useState(null);
+
+  const onCascaderChange = (_value) => {
+    onChange(selectorIndices.TUTOR_ASSIGNMENT);
+  };
+
+  useEffect(() => {
+    quarterProvider()
+      .then(quarters => {
+        setOptions(quarters.map(q => ({
+          value: q,
+          label: asHumanQuarter(q),
+          isLeaf: false
+        })));
+      })
+  }, [quarterProvider]);
+
+  const loadData = async selectedOptions => {
+    if (selectedOptions.length === 1) {
+      const selectedQuarter = selectedOptions[0];
```

Comment:
For consistency, we should use the selectorIndices here
```suggestion
      const selectedQuarter = selectedOptions[selectorIndices.QUARTER];
```